dbt/include/athena/macros/adapters/relation.sql

```diff
@@ -1,8 +1,30 @@
 {% macro athena__drop_relation(relation) -%}
+  {%- set native_drop = config.get('native_drop', default='False') | as_bool -%}
+
+  {%- if native_drop -%}
+    {%- do drop_relation_sql(relation) -%}
```

Comment:
Native drops works only for iceberg, this means if a user use the native drop globally and it uses hive tables, data in s3 won't be clean.
We should add another condition in case of native drop that check the relation type, if we deal with iceberg tables or views we execute only drop, if the relation is a table we want to clean up also s3.

Reply:
Addressed